### PR TITLE
Use trigger and condition type translation labels

### LINKED ISSUE #18347: Missing in lokalise existing automation - "Calendar trigger"
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Missing in lokalise

Screen can be found in web version Settings/Automations and Scenes/ in existing automatisation

Snag_CalendarTrigger

### Describe the behavior you expected

Add to lokalise

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

2023.10.5

### What was the last working version of Home Assistant Core?

not relevant

### In which browser are you experiencing the issue with?

_No response_

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### LINKED ISSUE #18349: Missing in lokalise existing automation - "Device trigger"
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Missing in lokalise

Screen can be found in web version Settings/Automations and Scenes/ in existing automatisation

Snag_DeviceTrigger

### Describe the behavior you expected

Add to lokalise

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

2023.10.5

### What was the last working version of Home Assistant Core?

not relevant

### In which browser are you experiencing the issue with?

_No response_

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### LINKED ISSUE #18350: Missing in lokalise existing automation - "Event trigger"
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Missing in lokalise

Screen can be found in web version Settings/Automations and Scenes/ in existing automatisation

Snag_EventTrigger

### Describe the behavior you expected

Add to lokalise

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

2023.10.5

### What was the last working version of Home Assistant Core?

not relevant

### In which browser are you experiencing the issue with?

_No response_

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### LINKED ISSUE #18351: Missing in lokalise existing automation - "Geo location trigger"
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Missing in lokalise

Screen can be found in web version Settings/Automations and Scenes/ in existing automatisation

Snag_GeoLocationTrigger

### Describe the behavior you expected

Add to lokalise

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

2023.10.5

### What was the last working version of Home Assistant Core?

not relevant

### In which browser are you experiencing the issue with?

_No response_

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### LINKED ISSUE #18353: Missing in lokalise existing automation - "Numeric state condition"
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Missing in lokalise

Screen can be found in web version Settings/Automations and Scenes/ in existing automatisation

Snag_NumericStateCondition

### Describe the behavior you expected

Add to lokalise

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

2023.10.5

### What was the last working version of Home Assistant Core?

not relevant

### In which browser are you experiencing the issue with?

_No response_

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### LINKED ISSUE #18359: Missing in lokalise existing automation - "Zone Trigger"
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Missing in lokalise

Screen can be found in web version Settings/Automations and Scenes/ in existing automatisation

Snag_ZoneTrigger

### Describe the behavior you expected

Add to lokalise

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

2023.10.5

### What was the last working version of Home Assistant Core?

not relevant

### In which browser are you experiencing the issue with?

_No response_

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by silamon**
## Proposed change
Use existing translation labels to display the trigger or condition type. 
To not add a few more labels, I've removed the redundant "trigger" or "condition" mentioned after the type. 

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #18359 fixes  #18353 fixes #18351 fixes #18350 fixes #18349 fixes #18347
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/data/automation_i18n.ts [hunk 1]
```diff
@@ -661,9 +658,16 @@ const tryDescribeTrigger = (
...
+  const triggerType = trigger.platform
+    ? hass.localize(
+        `ui.panel.config.automation.editor.triggers.type.${trigger.platform}.label`
+      )
+    : "";
+
+  return (
+    triggerType ||
+    hass.localize(`ui.panel.config.automation.editor.triggers.unknown_trigger`)
+  );
```
Comment (bramkragten) [inline src/data/automation_i18n.ts:670]:
```suggestion
  return (
    hass.localize(
        `ui.panel.config.automation.editor.triggers.type.${trigger.platform}.label`
      ) ||
    hass.localize(`ui.panel.config.automation.editor.triggers.unknown_trigger`)
  );
```

File: src/data/automation_i18n.ts [hunk 2]
```diff
@@ -1093,14 +1094,20 @@ const tryDescribeCondition = (
...
-    condition.condition ? condition.condition.replace(/_/g, " ") : "Unknown"
-  } condition`;
+  const conditionType = condition.condition
+    ? hass.localize(
+        `ui.panel.config.automation.editor.conditions.type.${condition.condition}.label`
+      )
+    : "";
+
+  return (
+    conditionType ||
```
Comment (bramkragten) [inline src/data/automation_i18n.ts:1108]:
```suggestion

  return (
    hass.localize(
        `ui.panel.config.automation.editor.conditions.type.${condition.condition}.label`
      ) ||
```

Comment (bramkragten) [reply src/data/automation_i18n.ts:1108]: Or if it was done for typing issues:
```ts
return condition.condition ?
hass.localize(`ui.panel.config.automation.editor.conditions.type.${condition.condition}.label`)
:  hass.localize(
      `ui.panel.config.automation.editor.conditions.unknown_condition`
    )
```

**Comment by silamon** [comment]
There were 3 cases to handle: 
1) an existing trigger/condition has been given => show translation
2) an unexisting trigger/condition has been given => show unknown
3) no type is provided, thus unexisting trigger/condition => show unknown 

So your suggestions were fine to add.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**